### ISSUE TITLE
effect of trimming

### ISSUE THREAD

**Issue by cresil** [benjjneb/dada2#296]
Currently im working on a 16S (V34) data set of rather poor quality. In my initial dada2 analyses I did not trim the poor quality sections but rather allowed for more mismatches in the overlap while merging. This left me with an inflated sequence table (10k forward, 1.2k reverse, 150k merged sequence variants). 

So i decided to trim the raw data (300> 265*215 resp) and rerun dada2. To my surprise the obtained sequence table was even more inflated (13k, 4k and 400k).  

Comparing the two sets showed that only 6k sequence variants out of 150k and 400k matched, representing roughly 70% of the data. 

So this leaves me with a few questions. What causes the inflation of the final seqtab? I guess it has something to do with the fact that the number of singletons decreases with trimming and therefore more partitions will be able to form. I also noticed that the error models show a much higher error frequency for non trimmed data compared to the trimmed data.

So, in short, if trimming has such an effect on the output, what approach is more reliable?

**Comment by benjjneb** [comment]
The dada2 algorithm relies on repeated observations of identical sequences to call biological sequences. When sequences have a very low probability of being error-free over the whole read (as is the case when very low-quality sequence tails are included) the algorithm will fail to identify low-frequency variants, because there is only 1 or 0 error-free reads representing those variants. When you trimmed the tails, you recovered more of those low frequency variants, thus the higher number of sequences. This also shows up in the error model, which will actually be more accurate on the trimmed date for the reason above (error learning involves inferring the sample composition).

Per the above, we recommend trimming reads at the point where quality scores crash, with the recognitions that this has to be balanced with the need to maintain overlap between reads in order to merge them later. I discuss this a bit more here: https://github.com/benjjneb/dada2/issues/236#issuecomment-298197310

On your data specifically: When running the trimmed (and quality filtered?) data, did you return to the standard recommendation of allowing no mismatches in the overlap region? And are your numbers of SVs from before or after chimera filtering (chimera formation can be high in multi-V-region amplicons because of the conserved region in the middle)?

**Comment by cresil** [comment]
Thanks for the clarification. I kept the allowance of mismatches in the second (trimmed) run. Setting it to zero reduced the amount of ASVS to ~2000. I did lose roughly 25 % of the data, Chimera removal removed another ~10%. Losing so much data does give me this uneasy feeling. 

However looking at my mock things look very decent. 44 out of 55 members have a perfect ASV representing them out of a total of 110 ASV, the others mock have an ASV 1-3 off. The other ASV are chimeras, possible cross contamination and possible natural sequence variants in the mock. 

Thanks!

**Comment by benjjneb** [comment]
That is a reasonable amount of data to lose to filtering/merging on a multi-V-region amplicon where the low quality tails come into play, and the number of chimeras also looks reasonable.

Given that your mock quality control looks good (and kudos to you for sequencing that and making use of it!) I would feel pretty good about your data now.
